packages/services/f-braze-adapter/src/services/utils/locationFilter.js

```diff
@@ -19,6 +30,10 @@ const locationFilter = (cards, { location }) => cards.filter(card => {
         if (!location) {
             return false;
         }
+        // if we don't have a valid lat long throw it out
+        if ((needsLongitude && needsLatitude) && !hasValidLocation(latitude, longitude)) {
```

Comment:
I don't believe it's guaranteed that a lat or lon will be required to also have a location, in which case this should also be outside of the `if (needsLocation) {` block. Also, this should be an OR condition rather than AND on the (admittedly odd, but possible) chance that a string has included a latitude placeholder without a longitude placeholder, or vice-versa.

Reply:
On the first point it may not require a location, but do we want to filter out any that do not have a location KVP? My understanding was we would filter out any that have invalid lat, long if they have a location? Second point good catch will make that change.

Reply:
Yes, good point - but we shouldn't make that dependent on also having a location placeholder as well. I.e. we might have a url `www.example.com/$LAT$/$LON$` that doesn't have a reference to the location, but we should allow that to show (and the placeholders populated accordingly) if the `location` KVP matches to the last stored location.